dev/com.ibm.ws.security.jwt/src/com/ibm/ws/security/jwt/internal/JwtCache.java

```diff
@@ -23,33 +22,37 @@
 import com.ibm.websphere.ras.TraceComponent;
 import com.ibm.websphere.ras.annotation.Sensitive;
 import com.ibm.ws.ffdc.annotation.FFDCIgnore;
-import com.ibm.ws.security.common.structures.CacheEntry;
 import com.ibm.ws.security.common.structures.SingleTableCache;
-import com.ibm.ws.security.jwt.config.JwtConsumerConfig;
 
 public class JwtCache extends SingleTableCache {
 
     private static final TraceComponent tc = Tr.register(JwtCache.class);
 
     static final int DEFAULT_ENTRY_LIMIT = 500;
 
-    private final JwtConsumerConfig config;
-
-    public JwtCache(long timeoutInMilliSeconds, JwtConsumerConfig config) {
+    public JwtCache(long timeoutInMilliSeconds) {
         super(DEFAULT_ENTRY_LIMIT, timeoutInMilliSeconds);
-        this.config = config;
     }
 
     /**
      * Find and return the object associated with the specified key.
      */
-    @Override
-    public synchronized Object get(@Sensitive String key) {
-        JwtContext jwtContext = (JwtContext) super.get(key);
-        if (jwtContext == null || isJwtExpired(jwtContext)) {
+    public synchronized Object get(@Sensitive String jwt, String configId) {
+        JwtCacheKey key = getCacheKey(jwt, configId);
+        JwtCacheValue cacheValue = (JwtCacheValue) get(key);
+        if (cacheValue == null || isJwtExpired(cacheValue)) {
```

Comment:
I think this method should simply be:
```suggestion
    public Object get(@Sensitive String jwt, String configId) {
        JwtCacheKey key = getCacheKey(jwt, configId);
        return super.get(key);
    }
```

Note that I took the synchronized off since the get method is already synchronized creating the key doesn't need to be synchronized.